temporal/t.rast.colors/t.rast.colors.py

```diff
@@ -147,44 +148,39 @@ def main():
 
     if rows:
         # Create the r.colors input file
-        filename = gs.tempfile(True)
-        file = open(filename, "w")
-
-        for row in rows:
-            string = "%s\n" % (row["id"])
-            file.write(string)
-
-        file.close()
-
-        flags_ = ""
-        if remove:
-            flags_ += "r"
-        if write:
-            flags_ += "w"
-        if list:
-            flags_ += "l"
-        if invert:
-            flags_ += "n"
-        if log:
-            flags_ += "g"
-        if abslog:
-            flags_ += "a"
-        if equi:
-            flags_ += "e"
-
-        try:
-            gs.run_command(
-                "r.colors",
-                flags=flags_,
-                file=filename,
-                color=color,
-                raster=raster,
-                volume=volume,
-                rules=rules,
-                overwrite=gs.overwrite(),
-            )
-        except CalledModuleError:
-            gs.fatal(_("Error in r.colors call"))
+        with tempfile.NamedTemporaryFile(mode="w", delete=True) as temp_file:
+            temp_file.write("\n".join(str(row["id"]) for row in rows))
+            temp_file.flush()
+
+            flags_ = ""
+            if remove:
+                flags_ += "r"
+            if write:
+                flags_ += "w"
+            if list:
+                flags_ += "l"
+            if invert:
+                flags_ += "n"
+            if log:
+                flags_ += "g"
+            if abslog:
+                flags_ += "a"
+            if equi:
+                flags_ += "e"
```

Comment:
If you move the `with` just below these lines, then there is almost no diff. The `flags_` doesn't use the ressource of the context manager, and is needed inside the call. It makes the context manager closer to the ressource use

Reply:
It looks great after your change!